src/BlackLinter.php

```diff
@@ -40,14 +43,71 @@ public function getLinterConfigurationName() {
     return 'black';
   }
 
+  public function getLinterConfigurationOptions() {
+    $options = array(
+      'black.python' => array(
+        'type' => 'optional string',
+        'help' => pht('Python version requirement.'),
+      ),
+    );
+
+    return $options + parent::getLinterConfigurationOptions();
+  }
+
+  public function setLinterConfigurationValue($key, $value) {
+    switch ($key) {
+      case 'black.python':
+        $this->pythonVersion = $value;
+        return;
+    }
+
+    return parent::setLinterConfigurationValue($key, $value);
+  }
+
   public function getLinterPriority() {
     return 0.01;
   }
   public function getDefaultBinary() {
     return 'black';
   }
 
+  private function checkVersion($version, $compare_to) {
+    $operator = '==';
+
+    $matches = null;
+    if (preg_match('/^([<>]=?|=)\s*(.*)$/', $compare_to, $matches)) {
+      $operator = $matches[1];
+      $compare_to = $matches[2];
+      if ($operator === '=') {
+        $operator = '==';
+      }
+    }
+
+    return version_compare($version, $compare_to, $operator);
+  }
+
+  private function getPythonVersion($cmd) {
+    list($err, $stdout, $stderr) = exec_manual('%C --version', $cmd);
+    return trim(str_replace('Python', '', $stdout));
+  }
+
   public function getVersion() {
+    if (!empty($this->pythonVersion)) {
+      $foundVersion = $this->getPythonVersion('python3');
```

Comment:
why pass in a cmd? will the cmd ever not be `'python3'`?

Reply:
I left the `getPythonVersion()` function general in case we ever want to try multiple possible interpreters.